test/integration/hydrodynamics.cc

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <gz/msgs/double.pb.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "gz/sim/Link.hh"
+#include "gz/sim/Model.hh"
+#include "gz/sim/Server.hh"
+#include "gz/sim/SystemLoader.hh"
+#include "gz/sim/TestFixture.hh"
+#include "gz/sim/Util.hh"
+#include "gz/sim/World.hh"
+
+#include "test_config.hh"
+#include "../helpers/EnvTestFixture.hh"
+
+#define PI 3.141592653
+
+using namespace gz;
+using namespace sim;
+
+class HydrodynamicsTest : public InternalFixture<::testing::Test>
+{
+  /// \brief Test a world file
+  /// \param[in] _world Path to world file
+  /// \param[in] _namespace Namespace for topic
+  /// \param[in] _density Fluid density
+  /// \param[in] _viscosity Fluid viscosity
+  /// \param[in] _radius Body's radius
+  /// \param[in] _area Body surface area
+  /// \param[in] _drag_coeff Body drag coefficient
+  public: void TestWorld(const std::string &_world,
+    const std::string &_namespace, double _density,
+     double _viscosity, double _area, double _drag_coeff);
+};
+
+//////////////////////////////////////////////////
+void HydrodynamicsTest::TestWorld(const std::string &_world,
+    const std::string &_namespace, double _density, double _viscosity,
+    double _area, double _drag_coeff)
+{
+  // Start server
+  ServerConfig serverConfig;
+  serverConfig.SetSdfFile(_world);
+
+  TestFixture fixture(serverConfig);
+
+  Model model;
+  Link body;
+  std::vector<math::Vector3d> bodyVels;
+  double dt{0.0};
+  fixture.
+  OnConfigure(
+    [&](const gz::sim::Entity &_worldEntity,
+      const std::shared_ptr<const sdf::Element> &/*_sdf*/,
+      gz::sim::EntityComponentManager &_ecm,
+      gz::sim::EventManager &/*eventMgr*/)
```

Comment:
Avoid using the `gz` and `sim` namespaces while already `using` them (lines 40/41). This makes it easier to port the code across branches, because older ones are still in `ignition::gazebo`